src/sentry/eventstore/models.py

```diff
@@ -157,45 +180,354 @@ def get_minimal_user(self):
-        return super(Event, self).title
+
+        et = eventtypes.get(self.get_event_type())()
+        return et.get_title(self.get_event_metadata())
 
     @property
     def culprit(self):
         column = self.__get_column_name(Columns.CULPRIT)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).culprit
+        return self.data.get("culprit")
 
     @property
     def location(self):
         column = self.__get_column_name(Columns.LOCATION)
         if column in self._snuba_data:
             return self._snuba_data[column]
-        return super(Event, self).location
+        et = eventtypes.get(self.get_event_type())()
+        return et.get_location(self.get_event_metadata())
+
+    @classmethod
+    def generate_node_id(cls, project_id, event_id):
+        """
+        Returns a deterministic node_id for this event based on the project_id
+        and event_id which together are globally unique. The event body should
+        be saved under this key in nodestore so it can be retrieved using the
+        same generated id when we only have project_id and event_id.
+        """
+        return md5("{}:{}".format(project_id, event_id)).hexdigest()
+
+    # TODO (alex) We need a better way to cache these properties.  functools32
+    # doesn't quite do the trick as there is a reference bug with unsaved
+    # models. But the current _group_cache thing is also clunky because these
+    # properties need to be stripped out in __getstate__.
+    @property
+    def group(self):
+        from sentry.models import Group
+
+        if not self.group_id:
+            return None
+        if not hasattr(self, "_group_cache"):
+            self._group_cache = Group.objects.get(id=self.group_id)
+        return self._group_cache
+
+    @group.setter
+    def group(self, group):
+        # guard against None to not fail on AttributeError
+        # otherwise Django 1.10 will swallow it in db.models.base init, but
+        # consequently fail to remove from kwargs, and you'll get the red herring
+        # TypeError: 'group' is an invalid keyword argument for this function.
+        if group is not None:
+            self.group_id = group.id
+            self._group_cache = group
+
+    @property
+    def project(self):
+        from sentry.models import Project
+
+        if not hasattr(self, "_project_cache"):
+            self._project_cache = Project.objects.get(id=self.project_id)
+        return self._project_cache
+
+    @project.setter
+    def project(self, project):
+        if project is None:
+            self.project_id = None
+        else:
+            self.project_id = project.id
+        self._project_cache = project
+
+    def get_interfaces(self):
+        return CanonicalKeyView(get_interfaces(self.data))
+
+    @memoize
+    def interfaces(self):
+        return self.get_interfaces()
+
+    def get_interface(self, name):
+        return self.interfaces.get(name)
```

Comment:
Do these need to stay public ?

Reply:
Yes, they are used in various places right now